### PR TITLE
Enable nGraph Debug ci test.

### PR DESCRIPTION
**Pull request by rgesteve**

### CODE REVIEW

**Comment by jywu-msft** [comment]
any updates on this PR?
maybe we can just enable debug builds for now - is the patch all that's needed?
the pipeline yml changes/test fixes can be enabled at a later time.

File: cmake/patches/ngraph/ngraph_protobuf.patch [hunk 1]
```diff
@@ -11,4 +21,61 @@ index 32217f5d..f6de5e76 100644
  
  if (WIN32)
      ExternalProject_Add(
+@@ -77,19 +77,23 @@ endif()
+ set(Protobuf_INSTALL_PREFIX ${EXTERNAL_PROJECTS_ROOT}/protobuf)
```
Comment (snnn) [inline cmake/patches/ngraph/ngraph_protobuf.patch]: Is it the same protobuf lib used by onnxruntime?

File: cmake/patches/ngraph/ngraph_protobuf.patch [hunk 2]
```diff
@@ -1,5 +1,15 @@
+From 698b4ba80e1af2599ae7946278d453dbddcecb98 Mon Sep 17 00:00:00 2001
```
Comment (jywu-msft) [inline cmake/patches/ngraph/ngraph_protobuf.patch]: remove the email header portion?

**Comment by jywu-msft** [comment]
I noticed during testing that nGraph does not seem to work with debug_alloc
you'll need to disable it by adding to existing ifndef here
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/platform/windows/debug_alloc.h#L7
and
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/platform/windows/debug_alloc.cc#L15

**Comment by jywu-msft** [comment]
seems something is wrong with the patch?

         Performing patch step for 'project_ngraph'
##[error]CUSTOMBUILD(0,0): Error : corrupt patch at line 52
    53>CUSTOMBUILD : error : corrupt patch at line 52 [D:\16\b\Debug\project_ngraph.vcxproj]

**Pull request merged**